libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -597,20 +603,28 @@ uint DagManager::setDagBlockOrder(blk_hash_t const &new_anchor, uint64_t period,
 
 void DagManager::recoverDag() {
   if (pbft_chain_) {
-    blk_hash_t pbft_block_hash = pbft_chain_->getLastPbftBlockHash();
-    if (pbft_block_hash) {
-      PbftBlock pbft_block = pbft_chain_->getPbftBlockInChain(pbft_block_hash);
-      blk_hash_t dag_block_hash_as_anchor = pbft_block.getPivotDagBlockHash();
+    auto pbft_block_hash = pbft_chain_->getLastPbftBlockHash();
+    while (pbft_block_hash) {
```

Comment:
are you checking if this is != NULL_BLOCK_HASH here ? If yes, it might be better to write it explicitely